components/datetime/src/format/time_zone.rs

```diff
@@ -41,271 +67,53 @@ where
-                    IsoMinutes::Required,
-                    IsoSeconds::Never,
-                )?,
-                3 => time_zone_format.iso8601_format(
-                    w,
-                    time_zone,
-                    IsoFormat::UtcExtended,
-                    IsoMinutes::Required,
-                    IsoSeconds::Never,
-                )?,
-                4 => time_zone_format.iso8601_format(
-                    w,
-                    time_zone,
-                    IsoFormat::UtcBasic,
-                    IsoMinutes::Required,
-                    IsoSeconds::Optional,
-                )?,
-                5 => time_zone_format.iso8601_format(
-                    w,
-                    time_zone,
-                    IsoFormat::UtcExtended,
-                    IsoMinutes::Required,
-                    IsoSeconds::Optional,
-                )?,
-                _ => {
-                    return Err(Error::Pattern(PatternError::FieldLengthInvalid(
-                        FieldSymbol::TimeZone(zone_symbol),
-                    )))
-                }
-            },
-            fields::TimeZone::UpperX => match field.length.idx() {
-                1 => time_zone_format.iso8601_format(
-                    w,
-                    time_zone,
-                    IsoFormat::Basic,
-                    IsoMinutes::Optional,
-                    IsoSeconds::Never,
-                )?,
-                2 => time_zone_format.iso8601_format(
-                    w,
-                    time_zone,
-                    IsoFormat::Basic,
-                    IsoMinutes::Required,
-                    IsoSeconds::Never,
-                )?,
-                3 => time_zone_format.iso8601_format(
-                    w,
-                    time_zone,
-                    IsoFormat::Extended,
-                    IsoMinutes::Required,
-                    IsoSeconds::Never,
-                )?,
-                4 => time_zone_format.iso8601_format(
-                    w,
-                    time_zone,
-                    IsoFormat::Basic,
-                    IsoMinutes::Required,
-                    IsoSeconds::Optional,
-                )?,
-                5 => time_zone_format.iso8601_format(
-                    w,
-                    time_zone,
-                    IsoFormat::Extended,
-                    IsoMinutes::Required,
-                    IsoSeconds::Optional,
-                )?,
-                _ => {
-                    return Err(Error::Pattern(PatternError::FieldLengthInvalid(
-                        FieldSymbol::TimeZone(zone_symbol),
-                    )))
-                }
-            },
+    fn write_pattern<W>(
+        &self,
+        patterns: &DataPayload<PatternPluralsFromPatternsV1Marker>,
+        w: &mut W,
+    ) -> Result<(), Error>
+    where
+        W: fmt::Write + ?Sized,
+    {
+        let pattern = patterns
+            .get()
+            .0
+            .clone()
```

Comment:
Nit: We shouldn't need to clone this, because we're only iterating over it.

Reply:
I got an error if I remove `.clone()`:
```
error[E0507]: cannot move out of a shared reference
   --> components/datetime/src/format/time_zone.rs:112:23
    |
112 |           let pattern = patterns
    |  _______________________^
113 | |             .get()
114 | |             .0
    | |_________^ move occurs because value has type `PatternPlurals<'_>`, which does not implement the `Copy` trait
```